depthai_ros_driver/src/camera.cpp

```diff
@@ -0,0 +1,184 @@
+                daiNodes.push_back(std::move(nn));
+                break;
+            }
+            case param_handlers::camera::NNType::Spatial: {
+                RCLCPP_WARN(this->get_logger(), "Spatial NN selected, but configuration is RGB.");
+            }
+            default:
+                break;
+        }
+        daiNodes.push_back(std::move(rgb));
+    } else if(ph->getParam<std::string>(this, "i_pipeline_type") == "RGBD") {
+        auto rgb = std::make_unique<dai_nodes::Sensor>("color", this, pipeline,device, dai::CameraBoardSocket::RGB);
+        auto stereo = std::make_unique<dai_nodes::Stereo>("stereo", this, pipeline, device);
+        auto nn_type = ph->getNNType(this);
+        switch(nn_type) {
+            case param_handlers::camera::NNType::None:
+                break;
+            case param_handlers::camera::NNType::RGB: {
+                auto nn = std::make_unique<dai_nodes::NN>("nn", this, pipeline);
+                rgb->link(nn->getInput(), static_cast<int>(dai_nodes::link_types::RGBLinkType::preview));
+                daiNodes.push_back(std::move(nn));
+                break;
+            }
+            case param_handlers::camera::NNType::Spatial: {
+                auto nn = std::make_unique<dai_nodes::SpatialNN>("nn", this, pipeline);
+                rgb->link(nn->getInput(static_cast<int>(dai_nodes::nn_helpers::link_types::SpatialNNLinkType::input)), static_cast<int>(dai_nodes::link_types::RGBLinkType::preview));
+                stereo->link(nn->getInput(static_cast<int>(dai_nodes::nn_helpers::link_types::SpatialNNLinkType::inputDepth)));
+                daiNodes.push_back(std::move(nn));
+                break;
+            }
+            default:
+                break;
+        }
+        daiNodes.push_back(std::move(rgb));
+        daiNodes.push_back(std::move(stereo));
+    } else {
+        std::string configuration =
+            ph->getParam<std::string>(this, "i_pipeline_type") + " " + ph->getParam<std::string>(this, "i_cam_type");
+        throw std::runtime_error("UNKNOWN PIPELINE TYPE SPECIFIED/CAMERA DOESN'T SUPPORT GIVEN PIPELINE. Configuration: " + configuration);
+    }
+    if(ph->getParam<bool>(this, "i_enable_imu")) {
+        auto imu = std::make_unique<dai_nodes::Imu>("imu", this, pipeline);
+        daiNodes.push_back(std::move(imu));
+    }
+    device.reset();
+    RCLCPP_INFO(this->get_logger(), "Finished setting up pipeline.");
+}
+
+void Camera::setupQueues() {
+    for(const auto& node : daiNodes) {
+        node->setupQueues(device);
+    }
+}
+
+void Camera::startDevice() {
+    dai::DeviceInfo info;
+    auto mxid = ph->getParam<std::string>(this, "i_mx_id");
+    auto ip = ph->getParam<std::string>(this, "i_ip");
+    if(!mxid.empty()) {
+        RCLCPP_INFO(this->get_logger(), "Connecting to the camera using mxid: %s", mxid.c_str());
+        info = dai::DeviceInfo(mxid);
```

Comment:
https://github.com/luxonis/depthai-ros/blob/ros-release/depthai_examples/ros2_src/stereo_inertial_publisher.cpp#L416

Best to use [this](https://github.com/luxonis/depthai-ros/blob/ros-release/depthai_examples/ros2_src/stereo_inertial_publisher.cpp#L416) method here as it provides more information. 
That way we can also check if the device is already being connected by another program and provide direct warning. For ex: if the user is running a python example.

Reply:
updated starting method